### DISCUSSION TITLE
Can I make a static site with relative paths?

### DISCUSSION THREAD

**Discussion by unuseless** [datenstrom/community#472] [Ask a question] [answered]
Currently when Yellow creates a static site, the navigation and css become:
```
<link rel="stylesheet" type="text/css" media="all" href="/media/resources/bundle-d86ed368b7.min.css" />
```
and
```
<li><a href="/about/">About</a></li>
```
Is it possible to generate relative links, i.e. without the leading slash:
```
<link rel="stylesheet" type="text/css" media="all" href="media/resources/bundle-d86ed368b7.min.css" />
```
```
<li><a href="about/">About</a></li>
```

That allows the static site to be previewed locally without a webserver.

**Comment by markseuffert** [comment, marked as answer]
As far as I know, Datenstrom Yellow cannot generate links without a leading slash. 

For previewing a static website you can use the [built-in web server](https://github.com/annaesvensson/yellow-serve), this is supported out of the box. For viewing a static website from the local file system, you probably have to create your own extension and convert hyperlinks in a post build step.

Not sure if other static site generators support such a feature. Would be good to know.

**Comment by unuseless** [comment]
I think frozen-flask does support relative links, i.e. without a leading slash.

With the built-in webserver, would I have to use
```
php yellow.php serve /public
```
I understand that I can preview the live version, but how can I preview the built version?

**Comment by BsNoSi** [comment]
**Given:**

file `D://a/b/c/d/file-a.html` containing three linkt to file `D://temp/file-b.html`
~~~
<a href="/../../../../../temp/file_b.html">Link</a>
<a href="../../../../../temp/file_b.html">Link</a>
<a href="./../../../../../temp/file_b.html">Link</a>
~~~

All three versions open `file-b` with Chrome based browsers incl. Microsoft Edge.
None of them in Firefox.  (OS: Win10x64)

It is not a shortage of Datenstrom Yellow, it's a a browser deficit of Firefox (or browser benefit of Chrome & Co. – depending on the point of view).

**Comment by wunderfeyd** [comment]
Please note the local file access was restricted in Firefox due to security/privacy reasons. https://bugzilla.mozilla.org/show_bug.cgi?id=1500453

**Comment by BsNoSi** [comment]
I can not follow the arguments given in bugzilla. But I note, that this is an issue in bugzilla, which carries in it's name that it seems to be a bug, doesn't it? 
Therfore, I would decline fixing a Firefox bug by changing a behaviour of Yellow CMS which possibly may lead to misbehaviour in all other - in my eyes correctly working – browsers.

**Comment by wunderfeyd** [comment]
The restriction applies to secondary loads like fetches, XHTMLRequests and local storage. Simple file access should work. I see your paths are wrong in your example. Please correct them and it would work in Firefox too.

**Comment by annaesvensson** [comment]
Thanks for your feedback. I am afraid to say that we won't add relative paths, not unless there is more demand. We can't promise if we may do something like this later, however we definitely appreciate the feedback.

**Comment by unuseless** [comment]
Thanks for the consideration!
